### PR TITLE
[CMSP-386] update WordPress (Composer Managed) docs

### LINKED ISSUE #9003: Follow up clean up on WordPress Composer documentation
In reviewing https://github.com/pantheon-systems/documentation/pull/9002 we found many opportunities for improving and clarifying how WordPress and Composer can be used with Pantheon. @jazzsequence expects that'll be done most effectively in conjunction with upcoming engineering work.

### PR DESCRIPTION
**Pull request by jazzsequence**
Closes #9003

## Summary

* **[WordPress with Composer on Pantheon: Introduction](https://docs.pantheon.io/guides/wordpress-composer)** - adds info about Bedrock and adds that Composer can be used to manage WordPress core updates
* **[Create a Composer-managed WordPress Site with Bedrock](https://docs.pantheon.io/guides/wordpress-composer/wordpress-composer-managed)** - Updates proper location of customizations (`config/application.php`), notes the existence and purpose of `config/application.pantheon.php`, updates doc to current recommendations and adds general Composer best practices
* **[Create a Composer-managed WordPress Site with Terminus Build Tools](https://docs.pantheon.io/guides/wordpress-composer/create-wp-site-composer-ci-auto-test)** - notes that Build Tools does not support WordPress Composer Managed yet, minor formatting changes and link updates
* **[Create a Composer-managed WordPress Site with Integrated Composer](https://docs.pantheon.io/guides/wordpress-composer/wordpress-ic)** - Updates the title to note, specifically, that this is for managing Composer-based _custom upstreams_, updates links (recommend forking from WPCM rather than `wordpress-project`), notes maintenance recommendations, adds notes about `pantheon-systems/upstream-management` package and fixes `upstream-configuration` path links.

### Dependencies and Timing

**Release**:
- [x] When ready
- [ ] After date: $DATE

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/old-path/` => `/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)

### CODE REVIEW

**Comment by jazzsequence** [comment]
I wouldn't be opposed to shortening the last article's title to something like `Create a Composer-managed WordPress Custom Upstream` (since the Integrated Composer is sort of assumed).

**Review by pwtyler — APPROVED** [review]
LGTM pending review from @rachelwhitton and/or @scottbuscemi.

**Review by scottbuscemi — APPROVED** [review]
fyi @pwtyler good to merge in

**Pull request merged**
